components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/SystemDefaultClaimMetadataManager.java

```diff
@@ -0,0 +1,277 @@
+
+    @Override
+    public List<ExternalClaim> getExternalClaims(String externalClaimDialectURI, int tenantId)
+            throws ClaimMetadataException {
+
+        if (externalClaimDialectURI == null || StringUtils.isBlank(externalClaimDialectURI)) {
+            return null;
+        }
+
+        if (externalClaimDialectURI.equals(LOCAL_CLAIM_DIALECT_URI)) {
+            throw new ClaimMetadataException("Invalid external claim dialect URI: " + externalClaimDialectURI);
+        }
+
+        return claims.getOrDefault(externalClaimDialectURI, Collections.emptyList()).stream()
+                .map(ExternalClaim.class::cast)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public ExternalClaim getExternalClaim(String externalClaimDialectURI, String claimURI, int tenantId) throws ClaimMetadataException {
+
+        if (externalClaimDialectURI == null || StringUtils.isBlank(externalClaimDialectURI) || claimURI == null ||
+                StringUtils.isBlank(claimURI)) {
+            return null;
+        }
+
+        if (externalClaimDialectURI.equals(LOCAL_CLAIM_DIALECT_URI)) {
+            throw new ClaimMetadataException("Invalid external claim dialect URI: " + externalClaimDialectURI);
+        }
+
+        return claims.getOrDefault(externalClaimDialectURI, Collections.emptyList()).stream()
+                .filter(claim -> claimURI.equals(claim.getClaimURI()))
+                .map(ExternalClaim.class::cast)
+                .findFirst()
+                .orElse(null);
+    }
+
+    @Override
+    public List<Claim> getMappedExternalClaims(String localClaimURI, int tenantId) throws ClaimMetadataException {
+
+        if (localClaimURI == null || StringUtils.isBlank(localClaimURI)) {
+            return null;
+        }
+
+        List<Claim> mappedExternalClaims = new ArrayList<>();
+        for (Map.Entry<String, List<Claim>> entry : claims.entrySet()) {
+            if (LOCAL_CLAIM_DIALECT_URI.equals(entry.getKey())) {
+                continue;
+            }
+            List<Claim> externalClaims = entry.getValue().stream()
+                    .map(ExternalClaim.class::cast)
+                    .filter(claim -> localClaimURI.equals(claim.getMappedLocalClaim()))
+                    .map(Claim.class::cast)
+                    .collect(Collectors.toList());
+            mappedExternalClaims.addAll(externalClaims);
+        }
+        return mappedExternalClaims;
+    }
+
+    @Override
+    public boolean isMappedLocalClaim(String localClaimURI, int tenantId) throws ClaimMetadataException {
+
+        if (localClaimURI == null || StringUtils.isBlank(localClaimURI)) {
+            return false;
+        }
+
+        for (Map.Entry<String, List<Claim>> entry : claims.entrySet()) {
+            if (LOCAL_CLAIM_DIALECT_URI.equals(entry.getKey())) {
+                continue;
+            }
+            List<Claim> externalClaims = entry.getValue().stream()
+                    .map(ExternalClaim.class::cast)
```

Comment:
to avoid potential class cast exceptions, we can use a instanceof filter before casting

`.filter(claim -> claim instanceof ExternalClaim)
.map(claim -> (ExternalClaim) claim)`

Reply:
Fixed with ecc4327051a74cf340ca167fc5026149147e4caf